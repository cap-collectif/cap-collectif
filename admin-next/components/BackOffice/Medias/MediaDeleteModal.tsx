import { useState, type FC } from 'react'
import { useIntl } from 'react-intl'
import { Modal, Heading, CapUIModalSize, Button, Text, toast } from '@cap-collectif/ui'
import DeleteMediaAdminMutation from '@mutations/DeleteMediaAdminMutation'
import { mutationErrorToast } from '@shared/utils/mutation-error-toast'

interface MediaDeleteModalProps {
  onClose: () => void
  medias: string[]
  connectionName?: string
}

const MediaDeleteModal: FC<MediaDeleteModalProps> = ({ onClose, medias, connectionName }) => {
  const intl = useIntl()
  const [isSubmitting, setIsSubmitting] = useState(false)

  const onDeleteMedias = () => {
    setIsSubmitting(true)
    return DeleteMediaAdminMutation.commit({
      input: {
        ids: medias,
      },
      connections: [connectionName],
    })
      .then(() => {
        setIsSubmitting(false)
        toast({
          variant: 'success',
          content: intl.formatMessage({ id: 'media-deleted' }, { num: medias.length }),
        })
        onClose()
      })
      .catch(() => {
        setIsSubmitting(false)
        mutationErrorToast(intl)
      })
  }

  return (
    <Modal
      show
      onClose={onClose}
      ariaLabel={intl.formatMessage({ id: 'global.image_uploader.image.preview' })}
      size={CapUIModalSize.Md}
    >
      <Modal.Header>
        <Heading>{intl.formatMessage({ id: 'media-delete-confirm' }, { num: medias.length })}</Heading>
      </Modal.Header>
      <Modal.Body>
        <Text>
          {intl.formatMessage(
            { id: 'media-delete-helptext' },
            { num: medias.length, b: (...chunks) => <b>{chunks.join('')}</b> },
          )}
        </Text>
      </Modal.Body>
      <Modal.Footer>
        <Button variantSize="big" variant="secondary" variantColor="hierarchy" onClick={onClose}>
          {intl.formatMessage({ id: 'global.cancel' })}
        </Button>
        <Button
          variantSize="big"
          variant="primary"
          variantColor="danger"
          isLoading={isSubmitting}
          onClick={onDeleteMedias}
        >
          {intl.formatMessage({ id: 'global.delete' })}
        </Button>
      </Modal.Footer>
    </Modal>
  )
}

export default MediaDeleteModal
