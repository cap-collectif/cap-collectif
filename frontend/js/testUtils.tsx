import * as React from 'react'
import ReactDOM from 'react-dom'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { Provider as ReakitProvider } from 'reakit'
import { IntlProvider } from 'react-intl-redux'
import { RelayEnvironmentProvider } from 'react-relay'
import { CapUIProvider } from '@cap-collectif/ui'
import { BrowserRouter as Router } from 'react-router-dom'
import { theme } from '~/styles/theme'
import appStore from '~/stores/AppStore'
import type { FeatureFlagType } from '~relay/useFeatureFlagQuery.graphql'
import { FormProvider, useForm } from 'react-hook-form'

export const mockRandomValues = () => {
  global.Math.random = () => 0.5
}
export const enableFeatureFlags = (flags: Array<FeatureFlagType>) => {
  global.mockFeatureFlag.mockImplementation((flag: Array<FeatureFlagType>) => {
    if (flags.includes(flag)) {
      return true
    }

    return false
  })
}
export const disableFeatureFlags = () => {
  global.mockFeatureFlag.mockImplementation(() => false)
}
export const mockUrl = (url: string) => {
  // https://stackoverflow.com/questions/54021037/how-to-mock-window-location-href-with-jest-vuejs
  delete window.location
  // @ts-expect-error assign new URL(...) to window.location because property location is not writable
  window.location = new URL(url)
}
export const addsSupportForPortals = () => {
  // See: https://github.com/facebook/react/issues/11565
  // @ts-expect-error
  ReactDOM.createPortal = jest.fn(element => {
    return element
  })
}
export const clearSupportForPortals = () => {
  // @ts-expect-error
  ReactDOM.createPortal.mockClear()
}
type Props = {
  children: JSX.Element | JSX.Element[] | string
  store?: any
  useCapUIProvider?: boolean
}
export const MockProviders = ({ children, store = {}, useCapUIProvider = false }: Props) => {
  return (
    <Provider store={appStore(store)}>
      <Router>
        <IntlProvider>
          <ReakitProvider>
            {useCapUIProvider ? (
              <CapUIProvider>{children}</CapUIProvider>
            ) : (
              <ThemeProvider theme={theme}>{children}</ThemeProvider>
            )}
          </ReakitProvider>
        </IntlProvider>
      </Router>
    </Provider>
  )
}
type RelaySuspensFragmentTestProps = Props & {
  environment: any
}
export const RelaySuspensFragmentTest = ({
  children,
  environment,
  store = {},
  useCapUIProvider = false,
}: RelaySuspensFragmentTestProps) => {
  return (
    <MockProviders store={store} useCapUIProvider={useCapUIProvider}>
      <RelayEnvironmentProvider environment={environment}>
        <React.Suspense fallback="">{children}</React.Suspense>
      </RelayEnvironmentProvider>
    </MockProviders>
  )
}

export const FormWrapper = props => {
  const formMethods = useForm()

  return <FormProvider {...formMethods}>{props.children}</FormProvider>
}

export default MockProviders
