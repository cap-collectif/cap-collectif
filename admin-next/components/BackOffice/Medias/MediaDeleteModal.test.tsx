/* eslint-env jest */
import { render } from '@testing-library/react'
import MediaDeleteModal from './MediaDeleteModal'
import { addsSupportForPortals, clearSupportForPortals, MockProviders } from 'tests/testUtils'

describe('<MediaDeleteModal />', () => {
  beforeEach(() => {
    addsSupportForPortals()
  })

  afterEach(() => {
    clearSupportForPortals()
  })

  it('should render correctly', () => {
    const { asFragment } = render(
      <MockProviders>
        <MediaDeleteModal onClose={jest.fn} medias={['m1', 'm2', 'm3']} />
      </MockProviders>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
