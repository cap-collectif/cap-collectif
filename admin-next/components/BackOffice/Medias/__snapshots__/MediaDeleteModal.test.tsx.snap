// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaDeleteModal /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXGCL gHoLKQ toasts-container toasts-container--top"
  />
  <div
    aria-label="global.image_uploader.image.preview"
    aria-modal="true"
    data-dialog="true"
    id="id-g"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="sc-bcXGCL embsrb sc-fvEvfv cap-modal__overlay"
      style="opacity: 0;"
    >
      <div
        class="sc-bcXGCL guZZlV sc-fcCDYY kmYErD cap-modal"
        style="opacity: 0; transform: translateY(-20px) translateZ(0);"
      >
        <header
          class="sc-bcXGCL jJSzZW cap-modal__header"
        >
          <div
            class="sc-bcXGCL jevVsb"
          >
            <h2
              class="sc-bcXGCL YIciv cap-heading"
              id="modal-title"
              style="margin-left: 0px; margin-top: 0px;"
            >
              media-delete-confirm {"num":3}
            </h2>
          </div>
          <button
            aria-describedby="modal-title"
            class="sc-bcXGCL cNICVM"
            type="button"
          >
            <svg
              aria-hidden="false"
              class="sc-bcXGCL sc-oZHua JBVPO cCzERh cap-icon"
              fill="none"
              height="1em"
              role="img"
              viewBox="0 0 12 12"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 11 11 1M1 1l10 10"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="0.8"
              />
            </svg>
          </button>
        </header>
        <div
          class="sc-bcXGCL eqUaSq cap-modal__body"
        >
          <p
            class="sc-bcXGCL kxBQAH cap-text"
          >
            media-delete-helptext {"num":3}
          </p>
        </div>
        <footer
          class="sc-bcXGCL fMWhlA cap-modal__footer"
        >
          <button
            class="sc-bcXGCL sc-gGvGpy kAXCTC kpqYfq cap-button"
            style="margin-left: 0px; margin-top: 0px;"
            type="button"
          >
            global.cancel
          </button>
          <button
            class="sc-bcXGCL sc-gGvGpy kAXCTC qZtxX cap-button"
            type="button"
          >
            global.delete
          </button>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
